### PR TITLE
Fixed properties that contain escaped strings for windows builds

### PR DESCRIPTION
**Pull request by noseglasses**
Some properties define in Arduino build system specification files are used as compiler command line arguments.

Arguments that are specified for Unix-like systems, e.g.
```
'-DLITERAL_STRING="the_string"'
```
cause compiler errors on Windows. These must be replaced to become
```
"-DLITERAL_STRING=\"the_string\""
```
on Windows.

This PR adds the necessary replacement to extra_flags and usb_flags.

### CODE REVIEW

File: cmake/Platform/Core/BoardFlags/CompilerFlagsSetter.cmake [hunk 1]
```diff
@@ -1,4 +1,19 @@
 # ToDo: Comment
+
+function(_sanitize_quotes
+   cmd_line_var_
```
Comment (MrPointer) [inline cmake/Platform/Core/BoardFlags/CompilerFlagsSetter.cmake]: 2 questions:

1. Why is it defined on a separate line?
2. Why is there an underscore (_) at **the end** of the var's name? Is it your convention?

File: cmake/Platform/Core/BoardFlags/CompilerFlagsSetter.cmake [hunk 2]
```diff
@@ -1,4 +1,19 @@
 # ToDo: Comment
+
```
Comment (MrPointer) [inline cmake/Platform/Core/BoardFlags/CompilerFlagsSetter.cmake]: Could you please document it like all other functions?

**Review by MrPointer — CHANGES_REQUESTED** [review]
First of all, thanks yet again for contributing such great code!

Is it only necessary when setting `extra_flags` and/or `usb_flags`? Or may it be necessary for other cases already existing in the project/future code?
Currently I can't seem to grab the pattern correctly, It would help if you'd explain it a bit further (As in when should one use that function).

Besides, I left some notes at specific lines.

File: cmake/Platform/Core/BoardFlags/CompilerFlagsSetter.cmake (hunk 2 above)
Comment (noseglasses) [reply cmake/Platform/Core/BoardFlags/CompilerFlagsSetter.cmake]: Yes, I can document it. 

The line
```
# ToDo: Comment
```
is left over from the also uncommented function `set_board_compiler_flags`.

File: cmake/Platform/Core/BoardFlags/CompilerFlagsSetter.cmake (hunk 1 above)
Comment (noseglasses) [reply cmake/Platform/Core/BoardFlags/CompilerFlagsSetter.cmake]: In all my projects I use underscores to distinguish local variables from variables that are defined in parent scope. This only makes a difference in functions, though.

I will replace `cmd_line_var_` with `CMD_LINE_VARIABLE` to make it more conforming and verbose.

**Comment by noseglasses** [comment]
> Is it only necessary when setting extra_flags and/or usb_flags?

To be on the safe side, `_sanitize_quotes` should be applied to all other properties that might end up in command lines. As far as I can see this currently only affects `extra_flags` and `usb_flags`.

**Review by MrPointer — APPROVED** [review]

**Pull request merged**